### PR TITLE
BadImport: Add `newInstance` to `BAD_STATIC_IDENTIFIERS`

### PR DESCRIPTION
**Pull request by rickie**
We now almost have our own version of this check as well. Do we still want to upstream this one? I think it would still be a nice improvement.

### CODE REVIEW

File: core/src/test/java/com/google/errorprone/bugpatterns/BadImportTest.java [hunk 1]
```diff
@@ -44,6 +44,23 @@ public void positive_static_simpleCase() {
...
+    compilationTestHelper
+        .addSourceLines(
+            "Test.java",
+            "import static com.google.errorprone.CompilationTestHelper.newInstance;",
+            "import com.google.errorprone.CompilationTestHelper;",
+            "import com.google.errorprone.bugpatterns.BugChecker;",
+            "",
+            "class Test {",
+            "  private final CompilationTestHelper compilationTestHelper =",
+            "  // BUG: Diagnostic contains:",
```
Comment (Stephan202) [inline core/src/test/java/com/google/errorprone/bugpatterns/BadImportTest.java]:
```suggestion
            "      // BUG: Diagnostic contains: CompilationTestHelper.newInstance",
```

**Review by Stephan202 — APPROVED** [review]
Rebased and added a commit. Suggested commit message:
```
BadImport: flag static import of `newInstance` methods
```

Agree that it makes sense to open this PR upstream.

**Comment by rickie** [comment]
Thanks for the suggestion! Will squash the changes and open a PR!

**Comment by rickie** [comment]
Filed https://github.com/google/error-prone/pull/3703.

**Comment by rickie** [comment]
Fixed in: https://github.com/google/error-prone/pull/3703.

**Pull request closed without merging**